How to create a Machine Learning App using the FastAPI and deploying it to the Kubernetes cluster

Bravinsimiyu opened the pull request:
This fixes #2799
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

Bravinsimiyu commented:
Hello, @lalith1403 I have created another PR.
Did what @hectorkambow suggested.

Bravinsimiyu commented:
Kindly check and see if everything is okay

Bravinsimiyu commented:
Hello @lalith1403, what's the status of this article.

lalith1403 commented:
This is right at top of the queue and should be published within the coming week. There were a few delays due to merge conflicts. Definitely will take it up at priority

Bravinsimiyu commented:
thanks for the update
@lalith1403

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+
+```python
+df['nationality'].unique()
+```
````
Output?

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+### Checking if our data is balanced
+
+```python
+df.groupby('nationality')['names'].size()
````
Output?

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+df.groupby('nationality')['names'].size()
+```
+
+### Visualizing our data using the matplotlib library
````
Matplotlib... Proper noun... should be capitalized.

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+
+```python
+import matplotlib.pyplot as plt
+%matplotlib inline
````
If you are using magic commands, then do the following:
1. Initially, before beginning the explanation, inform the user that we will be working with Jupyter Notebooks.
2. Paste a link to Colab Notebook so that the reader can verify the results easily.

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
```diff
@@ -0,0 +1,564 @@
+
+### Vectorizing our features
+
+We will use a count vectorizer to transform our dataset.
```
Explain in 2-3 lines what each library and function is doing.

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+
+```python
+nb.score(x_test,y_test)
+```
````
Output?

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
```diff
@@ -0,0 +1,564 @@
+
+### Saving our model using joblib
+
+We will use joblib to save our model into a pickle file.
```
When referring to libraries, variables, etc. use the following... \`\`

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
```diff
@@ -0,0 +1,564 @@
+
+### Introduction to the FastAPI
+
+[FastAPI](https://fastapi.tiangolo.com/) is a new python-based web framework used to create Web APIs. FastAPi is fast when serving the application; this enhances the app's performance.
```
Python

lalith1403 replied:
The first sentence is repeated here.

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+Use the following commands:
+
+```python
+pip install fast API
````
pip install fastapi

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+Let's install the server.
+
+```python
+pip install uvicorn[standard]
````
explain why we are using [standard] and not just `pip install uvicorn`

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+
+```python
+@app.get('/')
+async def index():
````
Explain the code. async is an advanced concept and needs to be explained in depth

lalith1403 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,564 @@
+
+```
+
+2. Creating docker layers
````
Explain briefly what docker layers are and how to create them.

lalith1403 reviewed (changes requested):
Hey @Bravinsimiyu. This is an amazing article and indeed brings a ton of value to the reader. Was very informative. Since this is a long article, there are quite a few minor changes. I have mentioned only a few occurrences. Please change all the occurrences of issues that are mentioned. Further, if you are using notebook, you need to specify that. Else, if using an editor, mention that. Since you are using a magic command, specific to notebooks, you need to consider this.

hectorkambow commented:
plagiarism check done

Bravinsimiyu commented:
Hello, @lalith1403 I have made the requested changes kindly review.

Bravinsimiyu commented:
Hello @lalith1403, so I have gone through all the conversations and added your suggestion. Also, I have added more explanations on some instances and linked a reader to a detailed article where necessary.
Some of the conversations were hidden earlier. Kindly review and see if I have applied all the necessary suggestions.

lalith1403 commented:
@ahmadmardeni1 @hectorkambow Kindly look into this article. There was a significant delay due to merge conflicts occurring at the time of the PR creation. This PR is older than a month. Kindly look into it whenever you get a chance. Thank you.

ahmadmardeni1 commented on content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md:
````diff
@@ -0,0 +1,807 @@
+
+Result:
+
+```bash
+e0f1bd4gv1f7t3dti5e89fd1o29341a50ete9hgad8ed0ye0ff27dt81667fu16b
+```
````
Is this a private key or something you should not share with anyone? @lalith1403

ahmadmardeni1 commented:
Great work gentlemen!

I spent a lot of time doing the final review for this article. @Bravinsimiyu 
- The paths are all wrong. And you have to crop the screenshots.
- A lot of repeated sentences, you should be precise. Otherwise, the reader will be bored.
Please go over the changes I made, and make sure to prevent the mistakes in your future articles.

Thank you for the review @lalith1403

lalith1403 commented:
Yes. Agreed. @Bravinsimiyu. I spent around an hour on this article. Please be sure to go through the changes made to avoid future article closures.

Bravinsimiyu commented:
Thanks, @lalith1403 and @ahmadmardeni1 I will go through them

Bravinsimiyu commented:
> Great work gentlemen!
> 
> I spent a lot of time doing the final review for this article. @Bravinsimiyu
> 
> * The paths are all wrong. And you have to crop the screenshots.
> * A lot of repeated sentences, you should be precise. Otherwise, the reader will be bored.
>   Please go over the changes I made, and make sure to prevent the mistakes in your future articles.
> 
> Thank you for the review @lalith1403

Thanks for the final review @ahmadmardeni1
